### DISCUSSION TITLE
unknown message: sw_interface_set_flags_f5aec1b8

### DISCUSSION THREAD

**Discussion by jinyan-0805** [FDio/govpp#52] [Q&A] [answered]
hi, I write a SetInterfaceState fuction to call SwInterfaceSetFlags, but respone that 
WARN[0008] Unable to retrieve message ID                 channel=1 error="unknown message: sw_interface_set_flags_f5aec1b8" msg_crc=f5aec1b8 msg_name=sw_interface_set_flags seq_num=4
E0929 18:10:39.615858 3441823 handler.go:427] set nic state failed: SetInterfaceState failed: req &{SwIfIndex:4 Flags:IF_STATUS_API_FLAG_ADMIN_UP} reply &{Retval:0}: unable to process request: unknown message: sw_interface_set_flags_f5aec1b8

```go
func (v *VppLink) SetInterfaceState(swIndex uint32, state int) error {
	klog.Infof("set interface %d state %d ", swIndex, state)
	v.lock.Lock()
	defer v.lock.Unlock()
	response := &interfaces.SwInterfaceSetFlagsReply{}
	request := &interfaces.SwInterfaceSetFlags{
		SwIfIndex: interface_types.InterfaceIndex(swIndex),
		Flags: interface_types.IfStatusFlags(state),
	}
	err := v.ch.SendRequest(request).ReceiveReply(response)
	if err != nil {
		return errors.Wrapf(err, "SetInterfaceState failed: req %+v reply %+v", request, response)
	}
	klog.Infof("set interface %d state %d success ", swIndex, state)
	return nil
}
```

**Comment by ondrej-fabry** [comment, marked as answer]
My guess is that [**version of VPP you are running**] is INCOMPATIBLE with the [**VPP version used to generate the binapi**] code you are using. If you are using the generated binapi code from the GoVPP repo then your VPP version must match the one specified at the top of the file (see package doc comment).

Try running [`CheckCompatibility`](https://github.com/FDio/govpp/blob/master/api/api.go#L96) method for the generated messages in a specific binapi package you are trying to use.


```go
if err := ch.CheckCompatiblity(interfaces.AllMessages()...); err != nil {
	log.Fatal(err)
}
```
> See full code in [simple-client example](https://github.com/FDio/govpp/blob/master/examples/simple-client/simple_client.go#L66-L71).

The error that is returned should tell you which messages are incompatible.
